aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1039,13 +1048,19 @@ impl AptosSimulationVM {
     fn validate_simulated_transaction<S: MoveResolverExt>(
         &self,
         session: &mut SessionExt<S>,
+        storage: &S,
```

Comment:
Do you really need this parameter everywhere? I thought there is a `session.get_data_store()`, and the resulting value contains the `S`. It might not be accessible, though?

Reply:
It expects Type instead of StructTag. I didn't see a particularly clean way to do this. The better option is probably to reveal a more useful dyn trait then this one in move. At which point we can clean this up. Although if you know a convenient way to go from struct tag to type, I'm game.

Reply:
I spent another hour looking into this...

To do it right:
* Update DataStore to expose a reader for getting a vec<u8> -> load_raw_resource
* Load the data_store from session
* Read the StructTag -> TypeTag and then ask TypeTag to Type from session
* Use load_raw_resource with type tag and source address
* Deserialize

That will likely be a several hour venture that improves code quality but at high cost and limited gain. I'd propose that as a good onboarding project for someone in the future.